### ISSUE TITLE
Paths: Follow on notes from internal usability tests

### ISSUE THREAD

**Issue by EDsCODE** [PostHog/posthog#6260]
I made them all TODO points but some might be wrapped into other changes.

Test 1
- [x] (https://github.com/PostHog/posthog/pull/6257)URLS jammed in dropdown
- [x] (https://github.com/PostHog/posthog/pull/6257) show full item on hover
- [ ] highlight all incoming paths on hover
- [ ] last node clarity
- [x] (https://github.com/PostHog/posthog/pull/6236)person modal title has step_1
- [ ] paths aren’t clear when leaving a node
- [ ] “continuing is confusing”
- [ ] cards are same column size from step 1 to step 2 when there are [a lot of cards](https://app.posthog.com/insights?insight=PATHS&end_point=insight%20viewed&step_limit=10&include_event_types=%5B%22custom_event%22%5D&path_groupings=%5B%5D&exclude_events=%5B%22insight%20loaded%22%5D&funnel_filter=%7B%7D)
- [ ] funnel to paths dropdown is overwhelming too many options that look similar
- [x] (https://github.com/PostHog/posthog/pull/6236) path didn’t respect funnel time frame
- [x] (https://github.com/PostHog/posthog/pull/6236) persons on funnel paths broken

Test 2
- [ ] end point for funnels isn’t pinned to the end which makes it confusing to see where all the paths are going
- [ ] end point should be destination flag or some indicator that within this context the person made it here but didn't necessarily drop off
- [ ] average time ambiguity. Is it from the root step or the step before?
- [ ] blank paths to start? and highlight 
- [x] (https://github.com/PostHog/posthog/pull/6236) funnel timestamp needs to match path timestamp
- [ ] default apply event filters when you navigate from funnels to paths
- [ ] actions somehow? Some companies (e.g. https://app.hubspot.com/contacts/6958578/company/6406150151) solely rely on actions/autocaptured events

**Important point**
The major outcome that we discussed with @marcushyett-ph is experimenting with the format of **not showing any path cards by default or just the start point and/or end points**. Then, only on hovering can you see the cards that are associated with an edge/path. This idea came up because with paths you're navigating using the relative sizes of the path connections, so exploring by hovering those edges will be less overwhelming than trying to sift through the forest of cards that immediately appear when there are a lot of paths. 

This will resolve most of the spacing/scaling issues that we're concerned about otherwise we would always run into some boundary with the maximum amount of information we can show. With this solution, we can even push the number of edges shown at once really aggressively because the cards won't overwhelm the visualization

@liyiy @clarkus

**Comment by clarkus** [comment]
This feedback makes sense, but I accounted for a lot of this in the figma designs. Is the goal here just to get to shipping and eventually roll out the changes to the visualization, or do we want to just rely on hover for showing cards? The solution in the designs is giving each path item a column of space to be fully described and then relying on scrolling or paging of some form to see subsequent steps. This would eliminate card overlaps. My concern with only showing this on hover is that it's not as immediately obvious what happened. If we could default to describing the largest (easiest to describe) items, that might make for a better first experience.

**Comment by EDsCODE** [comment]
Expanding column widths and scrolling will probably be required even with the hidden cards solution. And yea, big items by default would be a good start. The biggest problem I think this would solve is vertical congestion

**Comment by clarkus** [comment]
> Expanding column widths and scrolling will probably be required even with the hidden cards solution. And yea, big items by default would be a good start. The biggest problem I think this would solve is vertical congestion

The other change to make would be to increase the vertical spacing between path items in the same step column. This would at least mitigate the risk of cards overlapping.

**Comment by liyiy** [comment]
I think at least an option to toggle path card items display on/off hover could be nice, it seems like there's only so much shrinking/expanding of things we can do to the path cards and the paths visualization before it just won't scale. Another problem I've been seeing is that the path cards will hide the visualization underneath, perhaps we could also make the white more transparent?

**Comment by clarkus** [comment]
I think a good goal is to do the best job describing the top items in the visualization. A user can always adjust the query to focus on a specific start point to allow the visualization to fully describe that scenario. Below is a quick update from the paths design work that demonstrates this. The vis maxes out at showing 5 path step columns in our target viewport size. Larger viewports will see more steps, smaller viewports will have to scroll horizontally to see subsequent steps. For each path item we can show the step count, the name, total count, and menu icon. 

Cards are only attached to the top items - this is something to be defined. In this example, if the path item area was larger than the card, I showed the path card. If it was smaller / obscured by the card, I hid the card. In this case, those cards would be fully revealed on hover. When hovered, if the expanded card would overlap another card, display it above that card, or just hide the unrelated card (illustrated in step 4). 

https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=3471%3A29287

<img width="1174" alt="Screen Shot 2021-10-05 at 11 28 21 AM" src="https://user-images.githubusercontent.com/254612/136081595-af26784f-00c7-4d10-ba11-929f32419cd9.png">

@liyiy some background transparency is a good idea. I tried that early on and found it didn't do much to actually make the underlying vis more obvious. It mostly made the metrics slightly harder to read.
